File: create_tables/ba_address.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table ba_address';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table ba_address(
        BUSINESS_ASSOCIATE_ID VARCHAR(40) NOT NULL,
        SOURCE VARCHAR(40) NOT NULL,
        ADDRESS_OBS_NO INT NOT NULL,
        ACTIVE_IND VARCHAR(1),
        ADDRESSEE_TEXT VARCHAR(240),
        ADDRESS_TYPE VARCHAR(40),
        CITY_AREA_ID VARCHAR(40),
        CITY_AREA_TYPE VARCHAR(40),
        COUNTRY_AREA_ID VARCHAR(40),
        COUNTRY_AREA_TYPE VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        FIRST_ADDRESS_LINE VARCHAR(1024),
        OFFICE_TYPE VARCHAR(40),
        POSTAL_ZIP_CODE VARCHAR(40),
        PPDM_GUID VARCHAR(38),
        PREFERRED_IND VARCHAR(1),
        PROVINCE_STATE_AREA_ID VARCHAR(40),
        PROVINCE_STATE_AREA_TYPE VARCHAR(40),
        REMARK VARCHAR(2000),
        SECOND_ADDRESS_LINE VARCHAR(1024),
        SERVICE_PERIOD VARCHAR(240),
        SERVICE_QUALITY VARCHAR(40),
        THIRD_ADDRESS_LINE VARCHAR(1024),
        WITHHOLDING_TAX_IND VARCHAR(1),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/BaAddress.go
package dto

type Ba_address struct {
	Business_associate_id    string  `json:"business_associate_id" default:"source"`
	Source                   string  `json:"source" default:"source"`
	Address_obs_no           int     `json:"address_obs_no" default:"1"`
	Active_ind               *string `json:"active_ind" default:""`
	Addressee_text           *string `json:"addressee_text" default:""`
	Address_type             *string `json:"address_type" default:""`
	City_area_id             *string `json:"city_area_id" default:""`
	City_area_type           *string `json:"city_area_type" default:""`
	Country_area_id          *string `json:"country_area_id" default:""`
	Country_area_type        *string `json:"country_area_type" default:""`
	Effective_date           *string `json:"effective_date" default:""`
	Expiry_date              *string `json:"expiry_date" default:""`
	First_address_line       *string `json:"first_address_line" default:""`
	Office_type              *string `json:"office_type" default:""`
	Postal_zip_code          *string `json:"postal_zip_code" default:""`
	Ppdm_guid                *string `json:"ppdm_guid" default:""`
	Preferred_ind            *string `json:"preferred_ind" default:""`
	Province_state_area_id   *string `json:"province_state_area_id" default:""`
	Province_state_area_type *string `json:"province_state_area_type" default:""`
	Remark                   *string `json:"remark" default:""`
	Second_address_line      *string `json:"second_address_line" default:""`
	Service_period           *string `json:"service_period" default:""`
	Service_quality          *string `json:"service_quality" default:""`
	Third_address_line       *string `json:"third_address_line" default:""`
	Withholding_tax_ind      *string `json:"withholding_tax_ind" default:""`
	Row_changed_by           *string `json:"row_changed_by" default:""`
	Row_changed_date         *string `json:"row_changed_date" default:""`
	Row_created_by           *string `json:"row_created_by" default:""`
	Row_created_date         *string `json:"row_created_date" default:""`
	Row_effective_date       *string `json:"row_effective_date" default:""`
	Row_expiry_date          *string `json:"row_expiry_date" default:""`
	Row_quality              *string `json:"row_quality" default:""`
}
